waku/v2/peermanager/peer_manager.go

```diff
@@ -182,37 +157,45 @@ func (pm *PeerManager) getRelayPeers() (inRelayPeers peer.IDSlice, outRelayPeers
 	return
 }
 
-func (pm *PeerManager) connectToRelayPeers() {
+func (pm *PeerManager) ensureMinRelayConnsPerTopic() {
+	pm.topicMutex.RLock()
+	defer pm.topicMutex.RUnlock()
+	for topicStr, topicInst := range pm.subRelayTopics {
+		curPeers := topicInst.topic.ListPeers()
+		curPeerLen := len(curPeers)
+		if curPeerLen < relayOptimalPeersPerShard {
+			pm.logger.Info("Subscribed topic is unhealthy, initiating more connections to maintain health",
+				zap.String("pubSubTopic", topicStr), zap.Int("connectedPeerCount", curPeerLen),
+				zap.Int("optimumPeers", relayOptimalPeersPerShard))
+			//Find not connected peers.
+			notConnectedPeers := pm.getNotConnectedPers(topicStr)
+			if notConnectedPeers.Len() == 0 {
+				//TODO: Trigger on-demand discovery for this topic.
+				continue
+			}
+			//Connect to eligible peers.
+			numPeersToConnect := relayOptimalPeersPerShard - curPeerLen
 
+			if numPeersToConnect > notConnectedPeers.Len() {
+				numPeersToConnect = notConnectedPeers.Len()
+			}
+			pm.connectToPeers(notConnectedPeers[0:numPeersToConnect])
+		}
+	}
+}
+
+func (pm *PeerManager) connectToRelayPeers() {
 	//Check for out peer connections and connect to more peers.
-	inRelayPeers, outRelayPeers := pm.getRelayPeers()
-	pm.logger.Info("Number of Relay peers connected", zap.Int("inRelayPeers", inRelayPeers.Len()),
+	pm.ensureMinRelayConnsPerTopic()
+
+	inRelayPeers, outRelayPeers := pm.getRelayPeers(nil)
+	pm.logger.Info("Number of Relay peers connected",
+		zap.Int("inRelayPeers", inRelayPeers.Len()),
 		zap.Int("outRelayPeers", outRelayPeers.Len()))
```

Comment:
I think we can remove the `RelayPeers` suffix from here, as the log text already indicates these numbers refer to relay peers
```suggestion
	pm.logger.Info("number of relay peers connected",
		zap.Int("in", inRelayPeers.Len()),
		zap.Int("out", outRelayPeers.Len()))
```